### DISCUSSION TITLE
(Draft) R&D Grant Proposal for Shaga: AI Frame Interpolation

### DISCUSSION THREAD

**Discussion by chat-grp** [akash-network/community#580] [AI - Artificial Intelligence]
As some members of the Akash Community might have seen from [this quote tweet](https://x.com/gregosuri/status/1786881171013374131) from Greg Osuri, [Shaga ](https://twitter.com/Web3Shaga) has been using AKT after being slowed down for weeks by GCP and their unresponsiveness.

## The Ask:
We are asking for a grant worth $150'000 in AKT Tokens, with the condition that the tokens cannot be liquidated but only used to pay for compute on the Akash Network, similar to what the "Google Web3 Startup" program has granted us: $200'000 in GCP Credits.

This github discussion has the scope of getting feedback from the Akash community on milestones and distribution schedules.

## A brief intro about us:
Shaga is pioneering the P2P Cloud Gaming revolution, tapping into 1B+ idle Gaming PCs to unlock passive income for their owners and offer Game Streaming services to Gamers who want to simply play, with the lowest physically-possible latency. 

Our team is composed of a lethal mix of crypto-native distributed systems engineers and gaming industry veterans, with prominent figures from Xbox, Epic Games, Alienware and Riot Games.
We are on a mission to rebuild the gaming industry from the bottom-up, unifying all Gamers, Gaming PCs and Game Developers in a global distributed gaming network, allowing for self-sovereignty of both creators and consumers.

Our recent accomplishments:
- Won Solana Hyperdrive hackathon with proof of concept developed by solo dev in 1 month (me): https://x.com/solana/status/1720447734405062878
- Won Metaplex Hackathon with a solution around Blockchain-DRM that uses cNFTs on Solana to gate access to content: https://x.com/metaplex/status/1771573520042201337
- Got accepted in the Solana Incubator: https://x.com/solanalabs/status/1767949941731733930
- Got accepted in the Metaplex Startup program: https://x.com/metaplex/status/1788248706065842204

- Showcased mindblowing latency results, achieving what's probably the lowest-latency streaming protocol on the market at the moment (happy to receive challenges)
- Started our closed beta program for product feedback and heading towards an open beta launch soon

### Insight:
Success in AI primarily hinges on two critical factors: a substantial amount of high-quality data and extensive computational power to effectively minimize the loss function using that data. The distribution of play-time hours follows a power law, heavily favoring a few major titles such as Fortnite and Roblox, with Fortnite alone accumulating 96 billion hours of play by the end of 2020.

In video games, the "state space" refers to the entire set of distinct visual frames that can be generated from the game's fixed elements, such as textures, animations, and gameplay logic. This set is finite because the game, once released, operates within predetermined constraints that define what can occur within its environment. For example, the actions characters can perform, the environments they can interact with, and the outcomes of specific inputs are all pre-defined.
![image](https://github.com/akash-network/community/assets/128978529/748e534c-d804-4b93-8281-dc1b49c5cbf6)

Thus, the state space of a video game encapsulates all the possible frames a player might see, determined by the combination of all game elements under the rules set by the developers. This concept is crucial for technologies like frame interpolation, which aims to predict and fill in new frames to smooth transitions or improve frame rates, using the understanding of what frames are possible within the game's defined state space.
![image](https://github.com/akash-network/community/assets/128978529/3e624bfa-5fb8-41b2-91b1-035029948cd0)

Fascinated by Stephen Wolfram's insights over Computational Reducibility, an exciting idea emerged: with 96 billion hours of gameplay available, how can we analyze this data to enhance frame rates? Could there be a revolutionary breakthrough? In GameStreaming, frames are calculated with a half-round trip time (RTT) delay, as control input must reach the streaming device and then the frames are sent back with another half-RTT delay. 
![image](https://github.com/akash-network/community/assets/128978529/55bc2c00-ca5a-4425-80ce-c98a7108bc5a)

_**However, the control information originates from the client!!!**_ Therefore, with an efficient model, we could predict the trajectory of pixels based on inputs, having learned from state space data. For instance, every time you jump in Fortnite, the animation remains consistent; there's no need to recompute ray tracing for each frame to increase the Frame-Rate perceived by the Gamer
![image](https://github.com/akash-network/community/assets/128978529/8f8f813a-9dd0-4470-9159-c5d733dbb23b)

## Proposal Projected Outcomes:
(with a 25 AKT compute budget kindly donated by the Overclock Labs team to allow us to test Akash after being let down by GCP)

Currently FastRIFE is the lowest-latency model known to the scientific community: https://arxiv.org/pdf/2105.13482
What's notable here is that the frame generation time is between 26ms to 29ms, assuming it would be 30ms to simplify calculations, it means that **the Shaga Model optimized for GameStreaming could potentially "add" 33 Frames Per Second to the Gamer's Experience,** in the current state of things, considering we know in the AI & Computing Industry things tend to improve pretty quickly.
![image](https://github.com/akash-network/community/assets/128978529/ee4d3e7e-dd56-44dc-aafa-da1af0c46d5d)
On top of the better experience, this allows for reduced bandwidth consumption, with the trade-off of computational expenses on the client side.

We have been working on what we call "Future FastRIFE", will keep updating this proposal once we release paper and preliminary results.

## Proposal Timeline:
We expect the funds to be released at intervals based on achievements and milestones. These funds should cover our expenses for 1 year.

## Detail Impact:
This grant would significantly boost our ability to develop advanced AI models for Shaga's gaming network. As Shaga evolves towards profitability, we anticipate becoming a regular client of the Akash network, utilizing its robust computational resources for ongoing AI model training. 

This collaboration is crucial, given that Shaga's network predominantly relies on consumer GPUs, designed for gaming rather than intensive computing tasks. Access to Akash's superior computing infrastructure, equipped with high-performance GPUs like the A100s or H100s, will enable us to enhance the gaming experience significantly, which is not feasible with the standard consumer hardware found in most home setups.

**Comment by ghost** [comment]
Hey there, this thread is fascinating, sharing a few comments and observations 

If I understand correctly, the 150k USD in AKT is only to cover compute 
I mostly understand what you want to build, but I do not understand how we can bring this to the real world... are there gaming companies requesting this? or you plan on building this and then reach out to prospects 

Is this a service you would charge for? 
How do you plan on getting you and your team paid for this? 

Please add some links to your previous achievements to this thread so its easier for others to learn more about you 

Lastly, I'd suggest you jump in the sig-community Monthly Meeting, happening on May 14th at 11AM PST 

Google Meet joining info
Video call link: https://meet.google.com/sqt-uzhf-nba

**Reply by chat-grp** [reply]
hey thanks! will get on the call :)

**Comment by 31trainman** [comment]
I'd be interested in working with your team to build on this further with my teams current project to bring game servers to akash network with s3 backup capabilities. My discord is 31trainman
